packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -213,16 +212,22 @@ const ProductList = (props) => {
     const limitUrls = useLimitUrls()
 
     /**************** Action Handlers ****************/
+    const {data: wishlist} = useWishList()
     const addItemToWishlist = async (product) => {
         setWishlistLoading([...wishlistLoading, product.productId])
 
         // TODO: This wishlist object is from an old API, we need to replace it with the new one.
-        const listId = wishlist.data.id
-        const itemId = product.productId
-
-        createCustomerProductListItem(
+        const listId = wishlist.id
+        await createCustomerProductListItem(
```

Comment:
i don't think you need await here because there is nothing after this line in `addItemToWishlist`

Reply:
Every promise must _always_ be returned, awaited, or caught (`.catch`ed?). If you don't, you could end up with an uncaught rejection that will introduce subtle bugs and/or crash the app.

Reply:
yea, i mean to return the promise.